### PR TITLE
Oauth

### LINKED ISSUE #11: OAuth
We need to start implementing oauth and decide which services we will use. What this implies:

- [x] Set up passport in Express
- [x] Implement authenticated routes on the backend
- [x] Create appropriate accounts and get tokens

### PR DESCRIPTION
**Pull request by MynahMarie**
Closes #11 

In this PR I have:

- Implemented authentication with Facebook
- Set up Passport and FB strategy
- Implemented authenticated routes on the backend

### CODE REVIEW

**Review by hoslack — APPROVED** [review]

File: client/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "axios": "^0.17.1",
```
Comment (Karyum) [inline client/package.json]: let's see what happens #37

File: client/src/components/Navbar.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-  return (
-    <header className=" viewReportButton pv3 pv3-ns ph4-m ph5-l">
-      <nav className="f6 fw6 tc">
-        <a className=" f4 link br3 ph3 white" href="/" title="Home">
-          View Reports
-        </a>
-      </nav>
-    </header>
-  );
+    return (
```
Comment (Karyum) [inline client/src/components/Navbar.js]: you don't need the keyword return here cause this is an arrow function

File: client/src/components/Login.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+const Login = () => {
+    return (
```
Comment (Karyum) [inline client/src/components/Login.js]: arrow function here no need for return

File: client/src/components/AuthNavbar.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const AuthNavbar = () => {
+    return (
```
Comment (Karyum) [inline client/src/components/AuthNavbar.js]: and here :arrow_down:

File: client/src/components/ViewReports.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import AuthNavbar from './AuthNavbar';
+
+const ViewReports = () => {
+  return (
```
Comment (Karyum) [inline client/src/components/ViewReports.js]: :up:

File: client/src/config/dev.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+IMAGE_UPLOAD_PRESET: 'i50l2x2r',
+IMAGE_API_SECRET: 'S253wP6tc6hhr2G69DzzWdpi7Gw',
```
Comment (Karyum) [inline client/src/config/dev.js]: api keys !!!

File: server/routes/authRoutes.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    { scope: ['email'] }
+  )
+);
+
+authRouter.get('/google/callback', passport.authenticate('google'),
+  (req, res) => {
+    res.redirect('/viewreports');
+});
+
+authRouter.get('/current_user', requireLogin, (req, res) => {
```
Comment (Karyum) [inline server/routes/authRoutes.js:25]: i dont get this part, what is this route doing ?

File: server/models/Users.js [hunk 8]
```diff
@@ -2,9 +2,9 @@ const mongoose = require('mongoose');
 const { Schema } = mongoose;
 
 const UserSchema = new Schema({
+  strategy: String,
   auth_id: String,
-  email: String,
-  name: String,
+  name: String
```
Comment (Karyum) [inline server/models/Users.js:7]: cant find where you adding the name

**Review by Karyum — CHANGES_REQUESTED** [review]
# HOLY HELL backend on fleek

File: server/routes/authRoutes.js (hunk 7 above)
Comment (MynahMarie) [reply server/routes/authRoutes.js:25]: @Karyum this route checks if the user is logged in and redirects to the 'View Reports' page if he/she is...

File: client/src/components/Navbar.js (hunk 2 above)
Comment (MynahMarie) [reply client/src/components/Navbar.js]: @Karyum  Fixed!

File: client/src/components/Login.js (hunk 3 above)
Comment (MynahMarie) [reply client/src/components/Login.js]: @Karyum Fixed!

File: client/src/components/AuthNavbar.js (hunk 4 above)
Comment (MynahMarie) [reply client/src/components/AuthNavbar.js]: @Karyum Fixed!

File: client/src/components/ViewReports.js (hunk 5 above)
Comment (MynahMarie) [reply client/src/components/ViewReports.js]: @Karyum Fixed!

**Comment by Karyum** [comment]
@MynahMarie is this ready ?

**Comment by MynahMarie** [comment]
@Karyum Yes it is!! I did all the change you requested...

**Comment by Karyum** [comment]
@MynahMarie easy merge conflict, assign me after it was resolved

**Comment by MynahMarie** [comment]
@Karyum Ok fixed the merge conflicts!

**Review by Karyum — APPROVED** [review]

**Pull request merged**